Mtneer/mushu
# Firebase Authentication and Logout

# Description
All users should be authenticated in order to perform any activity in the system so that the system will be able to safely store user information, preferences, and budgets.
> 
> **Given** an unauthenticated user is in the Mushu application
> **When** they click any link
> **Then** they should be prompted to login using their email address
> 
> **Given** an unauthenticated user is viewing the login form
> **When** they enter an email address that matches an existing User Profile
> **Then** they should be authenticated into the system
> **And** they should be directed to the application home page
> 
> **Given** an unauthenticated user is viewing the login form
> **When** they enter an email address that does not match an existing User Profile
> **Then** an error message should be displayed
> **And** the user should be given another change to enter a valid email address

As a user I would like to be able to logout of the system so that I can ensure that no one else who uses my computer will have access to my Mushu account.
 
> **Given** an authenticated user is in the Mushu application
> **When** they select the `Logout` option from the menu
> **Then** they should be logged out of the system
> **And** they should be directed to the home page



## Type of change
Please delete options that are not relevant.

# Testing Instructions
- pull down the new branch
- if you haven't already, you'll need to set up your workspace. Run "npm install firebase", "npm install bootstrap", "npm install reactstrap" from the client directory. You'll need to go to firebase.google.com, create a new project titled "mushu", add two users "admin@admin.com" and "test@test.com", and copy the FirebaseProjectId into appsettings.json and the FirebaseWebAPIKey into .env.local.
- Once your environment is set up, start your web API from Visual Studio and start your react app from the console using "npm start" in the client directory
- The app should load to a Login page. Enter in "admin@admin.com" or "test@test.com" with the associated password. You should be logged in and navigated to a page that simply says "Hello".
- Click on Logout to check that the user is logged out and navigated back to the login page.

# Checklist:
